src/lib/txn.js

```diff
@@ -129,13 +129,19 @@ const sendSignedTransaction = (web3, stx, doubtNonceError) => {
   });
 };
 
-// returns a Promise<bool>, where the bool indicates tx success/failure
+// returns a Promise<void>, throwing on tx failure
 const waitForTransactionConfirm = (web3, txHash) => {
   return retry(async (bail, n) => {
     const receipt = await web3.eth.getTransactionReceipt(txHash);
     const confirmed = receipt !== null;
-    if (confirmed) return receipt.status === true;
-    else throw new Error('retrying');
+    if (!confirmed) {
+      throw new Error('Transaction not confirmed.');
+    }
+
+    const success = receipt.status === true;
+    if (!success) {
+      return bail(new Error('Transaction failed.'));
```

Comment:
agreed that it's very unclean 😢  I'll get better at this.

I was refactoring InviteEmails and we were checking the boolean response there, but in no other callsites (swallowing those errors) so I just made it do what we expected it to do.

Reply:
we probably expected it to throw because we're using it in a flow like

```
... construct transaction

... send transaction

... wait for transaction

return
```

all of which is (probably?) in a trycatch somewhere, and a failure at any point in that flow is treated as a failure of the flow as a whole. in every case where we'd check `if (success) {...}` we'd just throw a new error anyway

Reply:
> all of which is (probably?) in a trycatch somewhere,
> in every case where we'd check if (success) {...} we'd just throw a new error anyway

I suppose that's fair from a refactoring standpoint, yeah.